src/artifact/decl.rs

```diff
@@ -219,6 +219,24 @@ impl DefinedDecl {
         }
     }
 
+    /// Accessor to determine whether contents are executable
+    pub fn is_executable(&self) -> bool {
+        match self {
+            DefinedDecl::Section(a) => a.is_executable(),
+            DefinedDecl::Function(_) => true,
+            DefinedDecl::Data(_) => false,
```

Comment:
E.g., I think some JIT engines will want WX flags on data section? Or if i want to write self modifying code, etc., and have this in the data segments?

Reply:
We could make it a property of the Decl/DefinedDecl instead, I was just trying to make minimal changes. This may require either adding flags to every varient of Decl or converting DefinedDecl into a struct that contains an enum + flags or something?

Reply:
ah like something like:
```rust
/// A declaration that is defined inside this artifact
pub enum DefinedDeclKind {
    /// A function defined in this artifact
    Function(FunctionDecl),
    /// A data object defined in this artifact
    Data(DataDecl),
    /// A section defined in this artifact
    Section(SectionDecl),
}
pub struct DefinedDecl {
  kind: DefinedDeclKind,
  writable: bool,
  executable: bool,
}
```
?

Reply:
it's probably maybe a good idea (if those fields are truly shared amongst each of the Decl variants), but i suspect the diff might be quite large if DefinedDecl became a struct with a kind?

Reply:
the number of differences will be interesting, yes, but I'll put in the legwork if it's a good idea.

Reply:
Well it does worry me because:
1. it's a breaking change
2. I don't know who's using DefinedDecl, i think its effectively internal, but it would have been nice if it was marked `pub(crate)`
3. have a default value doesn't translate as well, because the default for a function decl is different than a data, etc.

At this point i think your PR is fine and we should just move forward with it.

Reply:
It is still possible to just add these flags to every varient, i.e. FunctionDecl, DataDecl, and SectionDecl, just like `datatype_methods!()`, and then the methods on DefinedDecl just delegate to the impl for the correct enum.

This wouldn't be a breaking change, right?

Reply:
Yes that wouldn't be a breaking change; i was just referring to changing the DefinedDecl to a struct; I think what you have now is basically done, modulo some of the comments I made, unless I'm missing something?

Reply:
so yes, i think this is (mostly?) done, unless you want me to add is_writable to FunctionDecl and/or is_executable to DataDecl.

we need to make sure not to forget making it work on mach, too.